src/app/projects/components/project-settings/components/data-schema/data-schema.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { attributeVisibilityStates, getTooltipVisibilityState } from '../../../create-new-attribute/attributes-visibility-helper';
+import { Attribute, AttributeVisibilityStates } from '../../entities/attribute.type';
+
+
+@Component({
+  selector: 'kern-data-schema',
+  templateUrl: './data-schema.component.html',
+  styleUrls: ['./data-schema.component.scss']
+})
+export class DataSchemaComponent implements OnChanges {
+
+  @Input() project: Project;
+  @Input() attributeVisibilityStates: AttributeVisibilityStates;
+  @Input() attributes: Attribute[];
+  @Input() pKeyValid: boolean;
+
+  tooltipsArray: string[] = [];
+
+  constructor(private projectApolloService: ProjectApolloService) {
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.attributes = changes.attributes?.currentValue;
+    this.pKeyValid = changes.pKeyValid?.currentValue;
```

Comment:
You don't need to set the values in the on changes, this will be done automatically. 

Also, this wont work since the changes only hold the values for the ones that did change. So with this approach if pkeyvalid is changed all other variables are set to null since they aren't in the structure. If we for some reason still would need to set them we could do something like:

```
if (changes.attributes) {
    this.attributes = changes.attributes.currentValue
}
```

However as mentioned in the beginning shouldn't be nessecary